File: src/Python/easy/math/problem2485.py

# First solution

class Solution:
    def pivotInteger(self, n: int) -> int:
        a = 0
        b = 0
        res = 0
        for i in range(n, 0, -1):
           a += i
           b = 0
           for j in range(1, i + 1):
               b += j
           if a == b:
               res = i
               break
        return res if res != 0 else -1

               
# Second solution

class Solution:
    def pivotInteger(self, n: int) -> int:
        a = sum(list(range(n + 1)))
        b = 0
        for i in range(n, 0, -1):
            a -= i
            if a == b:
                return i
            b += i
        return -1

n = 8
n = 4

solution = Solution()
solution.pivotInteger(n)

File: src/JavaScript/easy/math/problem2485.js
/**
 * @param {number} n
 * @return {number}
 */
var pivotInteger = function (n) {
  let a = 0;
  let b = 0;
  let res = null;
  for (let i = n; i > 0; i--) {
    a += i;
    b = 0;
    for (let j = 1; j <= i; j++) {
      b += j;
    }
    if (a == b) {
      res = i;
      break;
    } else {
      res = null;
    }
  }
  return res != null ? res : -1;
};

let n = 8;
// let n = 1
// let n = 4

pivotInteger(n);
